rts/idris_utf8.c

```diff
@@ -78,6 +78,36 @@ unsigned idris_utf8_index(char* s, int idx) {
    return top;
 }
 
+char* idris_utf8_advance(char* str, int i) {
+    // The pointer is not at the beginning of a character - find the next one
+    while ((*str & 0xc0) == 0x80) { str++; }
+
+    // Scan characters i times, but bail on end of string
+    while (i > 0 && *str != '\0') {
```

Comment:
Lets say a cstring is passed in of size N < 5, and the first byte tells the scanner to advance more than N bytes: This will pass by the check for `*str != '\0'` and overrun the buffer.